zone.go

```diff
@@ -452,44 +412,88 @@ func (api *API) ListZonesContext(ctx context.Context, opts ...ReqOption) (r Zone
 		return ZonesResponse{}, errors.Wrap(err, errUnmarshalError)
 	}
 
-	totalPageCount := r.TotalPages
+	if r.TotalPages < 2 {
+		return r, nil
+	}
+
+	// parameters of pagination
+	var (
+		totalPageCount = r.TotalPages
+		totalCount     = r.Total
+	)
+
+	// These checks should be redundant when CloudFlare works as advertised.
```

Comment:
i'm not sure i follow the comment here; what scenarios is this not the case?

Reply:
hmm, i think that is a little beyond the realm of this library. realistically, if Cloudflare responds with a 200, you should be safe to assume we're good for pagination parameters.

Reply:
I like to be a bit more defensive so that the code is logically `panic`-free. On the other hand, after checking the code more carefully, it's actually the case even without these checks, so I just removed them.

Reply:
The reason is that the following lines eliminate all possible conditions or memory access violations:

https://github.com/cloudflare/cloudflare-go/blob/7f2f0f88ecc2ea1f6c882428cb122007a0e88aa7/zone.go#L465-L472

Reply:
> i'm not sure i follow the comment here; what scenarios is this not the case?

I realized at least three test cases in `cloudflare_test.go` violated these checks.